src/shared/components/entry/TaxonomyView.tsx

```diff
@@ -25,17 +27,15 @@ export const TaxonomyId: FC<{ taxonId?: number }> = ({ taxonId }) => {
   return (
     <>
       <Link
-        to={generatePath(LocationToPath[Location.TaxonomyEntry], {
-          accession: `${taxonId}`,
-        })}
+        to={getEntryPath(Namespace.taxonomy, taxonId)}
       >{`${taxonId} `}</Link>
       <ExternalLink url={externalUrls.NCBI(taxonId)}>NCBI</ExternalLink>
     </>
   );
 };
 
 const TaxonomyView: FC<TaxonomyDataProps> = ({ data }) => {
-  if (!data) {
+  if (!data?.taxonId) {
```

Comment:
By checking only for `taxonId` here, we are making an assumption that the rest of the data being accessed below is safe, as long as the `taxonId` is present. Is that something we are confident will hold true always? I think the previous form which was checking for `!data` is more future proof.

Reply:
This check is more restrictive than before as it's both checking for no `data` (as before), and if there is, for no `data.taxonId`. Not sure how you want me to change it to be even more restrictive.